### PR TITLE
feat(YouTube): refactor and update selectors

### PR DESCRIPTION
**Pull request by theusaf**
## Description 
* Refactors the code
  * Removes lots of duplication of code
  * Uses `switch(true)` to replace long if-else chains
  * Moves various title/uploader queries into separate files/methods
  * Removes unneeded typing and other inconsistencies
* Fixes
  * Viewing a playlist now works
  * Potentially resolves viewing a video again?
* Notes/Observations
  * The old YouTube code may not be needed (new YouTube layout seems to be forced now?), but keeping it just in case.
  * Unsure if there is a difference between YouTube movies and videos in terms of query selectors

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

**Comment by DarkVillager** [comment]
Wow, nice code.

File: websites/Y/YouTube/video_sources/movies.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Resolver } from "../util";
+
+function isActive(): boolean {
+	return true;
```
Comment (Bas950) [inline websites/Y/YouTube/video_sources/movies.ts:4]:
```suggestion
	return !!getTitle() && !!getUploader();
```
Should it not be this?

**Review by Bas950 — APPROVED** [review]
For the rest LGTM

File: websites/Y/YouTube/presence.ts [hunk 2]
```diff
@@ -133,153 +106,49 @@ presence.on("UpdateData", async () => {
...
+			youtubeOldResolver,
+			youtubeTVResolver,
+			youtubeResolver,
+			youtubeMoviesResolver,
+		].find(resolver => resolver.isActive());
+
+		if (resolver === youtubeShortsResolver)
+			await cacheShortData(hostname, pathname.split("/shorts/")[1]);
+
+		const title = resolver.getTitle(),
```
Comment (Bas950) [inline websites/Y/YouTube/presence.ts:135]: isn't this resolver nullable? It might not find a resolver that is active?

**Review by Bas950 — APPROVED** [review]

**Review by Timeraa — APPROVED** [review]

**Pull request merged**